src/main/java/wolox/training/services/impl/BookServiceImpl.java

```diff
@@ -66,11 +67,17 @@ public Book createByOpenLibrary(String isbn) {
         }
     }
 
+    @Override
+    public List<Book> findByAllFields(String genre, String author, String image, String title,
```

Comment:
it's okay. I leave you only a comment, for clean code purposes you could recieve here a Request model class as parameter. This class could be built in the controller.

Reply:
Add Request model as parameter for the method